File: python/605_种花问题_easy.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if not flowerbed:
            return 0
        for i in range(len(flowerbed)):
            if n == 0:
                return True
            if flowerbed[i] == 0:
                if (i > 0 and flowerbed[i-1] == 1) or (i+1 < len(flowerbed) and flowerbed[i+1] == 1):
                    continue
                flowerbed[i] = 1
                n -= 1
        return n == 0

File: cpp/605_种花问题_easy.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int N = flowerbed.size();
        if(!N)
            return 0;
        for(int i=0; i<N; i++){
            if(n == 0)
                return true;
            if(flowerbed[i] == 0){
                if((i>0 && flowerbed[i-1] == 1) || (i+1<N && flowerbed[i+1] == 1))
                    continue;
                flowerbed[i] = 1;
                n--;
            }
        }
        return n==0;
    }
};